docs/application/dotnet/guides/nui/widget.md

```diff
@@ -0,0 +1,249 @@
+
+The following table lists the main functions to manage the widget instance:
+
+**Table: Main functions**
+| Function           | Description                                                                                   |
+|--------------------|-----------------------------------------------------------------------------------------------|
+| `OnCreate()`       | This function is called after the widget instance is created.                                 |
+| `OnTerminate()`    | This function is called after the widget instance is terminated.                              |
+| `OnPause()`        | This function is called when the widget is invisible.                                         |
+| `OnResume()`       | This function is called when the widget is visible.                                           |
+| `OnResize()`       | This function is called before the widget size is changed.                                    |
+| `OnUpdate()`       | This function is called when an event for updating the widget is received.                    |
+| `SetContentInfo()` | This function sends data about current state of widget instance to a widget viewer application. If you want to save the current state of widget before deleting it, you can transmit the current status to the viewer using SetContentInfo().|
+
+Most of functions are used for managing widget instance life-cycle.
+
+## Create a NUI Widget Application
+
+If you want to use NUI for your widget application, you must create the application with [NUI Widget Application](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.NUIWidgetApplication.html) class. 
+
+To create a NUI widget application:
+1. To use the methods and properties of [Tizen.NUI](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.html) namespace, include the following code:
+
+   ```csharp
+   using Tizen.NUI;
+   ```
+
+2. Define your widget application class, which is inherited from [NUIWidgetApplication](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.NUIWidgetApplication.html) class:
+
+   ```csharp
+   class Program : NUIWidgetApplication
+   ```
+3. The widget application starts with `Main()` function, which creates and initializes the application.`Run()` method of [NUIWidgetApplication](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.NUIWidgetApplication.html) class is used to start the application event-loop.
+
+   `NUIWidgetApplication` class provides two kinds of constructors:
+   - For using the `NUIWidgetApplication(Type type)` constructor, in case the widget application's ID is the same as the application ID.
+      ```csharp
+        static void Main(string[] args)
+        {
+            var app = new Program(typeof(MyWidget));
+            app.Run(args);
+        }    
+      ```
+
+   - For using `NUIWidgetApplication(Dictionary< Type, string > typeInfo) ` constructor, in case that your widget applications has multiple widget classes.
```

Comment:
@tscholb, suggest rephrasing to:

```suggestion
 - For using the `NUIWidgetApplication(Dictionary< Type, string > typeInfo) ` constructor, in case your widget applications have multiple widget classes.
```

Reference: https://github.com/Samsung/tizen-docs/pull/1052#discussion_r397356816